framework/src/kernels/NodalScalarKernel.C

```diff
@@ -12,27 +12,57 @@
 // MOOSE includes
 #include "Assembly.h"
 #include "MooseVariableScalar.h"
+#include "MooseMesh.h"
 #include "SystemBase.h"
 
 template <>
 InputParameters
 validParams<NodalScalarKernel>()
 {
   InputParameters params = validParams<ScalarKernel>();
-  params.addRequiredParam<std::vector<dof_id_type>>("nodes", "Node ids");
+  params.addParam<std::vector<dof_id_type>>("nodes", "Supply nodes using node ids");
+  params.addParam<std::vector<BoundaryName>>(
+      "boundary", "The list of boundary IDs  from the mesh where this nodal kernel applies");
+
   return params;
 }
 
 NodalScalarKernel::NodalScalarKernel(const InputParameters & parameters)
   : ScalarKernel(parameters),
     Coupleable(this, true),
     MooseVariableDependencyInterface(),
-    _node_ids(getParam<std::vector<dof_id_type>>("nodes"))
+    _node_ids(getParam<std::vector<dof_id_type>>("nodes")),
+    _boundary_names(getParam<std::vector<BoundaryName>>("nodeset_names"))
 {
   // Fill in the MooseVariable dependencies
-  const std::vector<MooseVariable *> & coupled_vars = getCoupledMooseVars();
+  const std::vector<MooseVariableFE *> & coupled_vars = getCoupledMooseVars();
   for (const auto & var : coupled_vars)
     addMooseVariableDependency(var);
+
+  // Check if node_ids and/or node_bc_names given
+  if ((_node_ids.size() == 0) && (_boundary_names.size() == 0))
+    mooseError("Must provide one of nodes or node_names");
+
+  if ((_node_ids.size() != 0) && (_boundary_names.size() != 0))
+    mooseError("Must provide only one input using nodes or node_names");
```

Comment:
"Both 'nodes' and 'boundary' parameters were specified. Use the 'boundary' parameter only."

Because we do nor want people to use nodes again.